androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/components/Settings.kt

```diff
@@ -0,0 +1,120 @@
+package com.profusion.androidenhancedvideoplayer.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.profusion.androidenhancedvideoplayer.R
+import com.profusion.androidenhancedvideoplayer.styling.Dimensions
+import kotlinx.coroutines.launch
+
+data class SettingsControlsCustomization(
+    val speeds: List<Float> = listOf(0.5f, 0.75f, 1.0f, 1.25f, 1.5f),
+    val speedIconContent: @Composable () -> Unit = { SpeedIcon() },
+    val modifier: Modifier = Modifier
+)
+
+@ExperimentalMaterial3Api
+@Composable
+fun Settings(
+    speed: Float,
+    onDismissRequest: () -> Unit,
+    onSpeedSelected: (Float) -> Unit,
+    customization: SettingsControlsCustomization
+) {
+    val sheetState = rememberModalBottomSheetState()
+
+    ModalBottomSheet(
+        onDismissRequest = onDismissRequest,
+        sheetState = sheetState,
+        modifier = customization.modifier.testTag("SettingsControlsParent")
+    ) {
+        SettingsSelector(
+            label = stringResource(id = R.string.settings_speed),
+            icon = customization.speedIconContent,
+            value = speed,
+            items = customization.speeds,
+            onSelected = onSpeedSelected
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T>SettingsSelector(
```

Comment:
nit: `private fun`, if we want to reuse somewhere else the developer will need to modify and refactor it into another component file